### ISSUE TITLE
Resumable upload is slow vs non-resumable upload

### ISSUE THREAD

**Issue by Landcross** [googleapis/google-api-python-client#793]
I have a problem where resumable uploads are significantly slower than non-resumable uploads.
I have a 1gbit internet connection. When I run non-resumable upload, python uploads the file at around 400-500 mbit. But when I upload the same file with `resumable=True` the upload speed is at most around 80-90 mbit.

#### Environment details

  - OS: Windows 10
  - Python version: 3.7
  - pip version: 19.0.3
  - `google-api-python-client` version: 1.7.11

#### Steps to reproduce
It's pretty straightforward. This is fast (I skipped the authentication portion of the test script, which is just copied from the quickstart):

```
media = MediaFileUpload(filename, mimetype=mimetype, resumable=False)
file = self.gdrive.files().create(body=file_metadata, media_body=media, fields='id').execute()
```

And this is slow:

```
media = MediaFileUpload(filename, mimetype=mimetype, resumable=True)
file = self.gdrive.files().create(body=file_metadata, media_body=media, fields='id').execute()
```

**Comment by busunkim96** [comment]
@Landcross it might be worth tweaking the chunk size to see if that makes a difference.

```
A MediaUpload for a file.

Construct a MediaFileUpload and pass as the media_body parameter of the
method. For example, if we had a service that allowed uploading images:

  media = MediaFileUpload('cow.png', mimetype='image/png',
    chunksize=1024*1024, resumable=True)
  farm.animals().insert(
      id='cow',
      name='cow.png',
      media_body=media).execute()

Depending on the platform you are working on, you may pass -1 as the
chunksize, which indicates that the entire file should be uploaded in a single
request. If the underlying platform supports streams, such as Python 2.6 or
later, then this can be very efficient as it avoids multiple connections, and
also avoids loading the entire file into memory before sending it. Note that
Google App Engine has a 5MB limit on request size, so you should never set
your chunksize larger than 5MB, or to -1.
```
https://googleapis.github.io/google-api-python-client/docs/epy/googleapiclient.http.MediaFileUpload-class.html

Issue #625 may be related.

**Comment by Landcross** [comment]
Thanks for the reply.

I have tried a few chunksizes and it doesn't make any real difference. I also tried -1 (thanks for the heads-up, at first I understood it wrong. I thought -1 would still load the whole file into memory), but that seems slow too. I'm not entirely 100% about the -1 though, I'll do some more testing tonight

As a sidenote: I'm not quite sure why it's mentioned that chunksize should not be larger than 5MB, while the default chunksize is `100 * 1024 * 1024` = 104857600 = 100MB.
